### PR TITLE
Adaptation of Ansible for Pi 5

### PR DESCRIPTION
**Pull request by alonsovidales**
This PR updates the build in order to work with the new Raspberry Pi 5.

We had to use the latest image since the old one only supports up to the Pi 4. The version of Moodle was also updated since with the new image we also get PHP 8.4.

Everything I tried seems to work without issues, but keep in mind that this involves ves a major bump in the PHP version, let's keep an eye just in case something start to fail.

#127

### CODE REVIEW

**Comment by ltoning** [comment]
Hi @alonsovidales did some testing on RPI 4 and Scratch didn't seems to work on my side. 
Not serving and no connection possible on 127:0.0.1:5801
All the rest look good.

**Comment by alonsovidales** [comment]
> Hi @alonsovidales did some testing on RPI 4 and Scratch didn't seems to work on my side. Not serving and no connection possible on 127:0.0.1:5801 All the rest look good.

Ups, I already gave my Pi4 to a friend... let me see if I can get it back to test it today 🤦‍♂️ thanks for catching this 🤗

**Comment by alonsovidales** [comment]
> Hi @alonsovidales did some testing on RPI 4 and Scratch didn't seems to work on my side. Not serving and no connection possible on 127:0.0.1:5801 All the rest look good.

Ok, it was this: https://github.com/ScratchAddons/ScratchAddons/issues/7456 I remember that I fixed manually bit I forget to update the ansible files 🤦‍♂️ updated 👍 thx!

**Comment by ltoning** [comment]
Hi @alonsovidales
 did some extensive testing on RPI 4.
 Bookworm comes with Networkmanager by default. This was interfering the the Wifi client feature we use for the ElimuPi-connect and  (maybe) the Wifi Hotspot in some way.

- Disabled NetworkManager  and reinstated wpa_supplicant.conf.
- Changed Wifi priority.

Can you do a new test on RPI 5?

**Review by ltoning — APPROVED** [review]

**Comment by alonsovidales** [comment]
> Hi @alonsovidales did some extensive testing on RPI 4. Bookworm comes with Networkmanager by default. This was interfering the the Wifi client feature we use for the ElimuPi-connect and (maybe) the Wifi Hotspot in some way.
> 
> * Disabled NetworkManager  and reinstated wpa_supplicant.conf.
> * Changed Wifi priority.
> 
> Can you do a new test on RPI 5?

Thanks for finding this out, but I'm not being able to reproduce it. In the 4 the wifi hotspot is working for me. how can I do to reproduce it? what did you do to make it work? 🤔

**Comment by ltoning** [comment]
Wifi client is not working. If you change wpa_supplicant.conf to match you
own wifi ssid you will notice it won't connect.
Hotspot works on uap0 works

Op zo 23 jun 2024 22:28 schreef Alonso Vidales ***@***.***>:

**Comment by alonsovidales** [comment]
Ahhh, ok, I was not aware of the ElimuConnect and 4G connections, sorry. I move them to NetworkManager, now it looks like:
```
***@***.***:~# nmcli connection show
NAME                UUID                                  TYPE      DEVICE
preconfigured       61f08a09-e3b4-461c-8a32-273b2cd83484  wifi      wlan0
lo                  85f3e588-bbf9-465b-9bbb-0816231f9ee0  loopback  lo
Wired connection 1  8ab2b7f0-2545-3145-9394-f2c0edbf7db9  ethernet  --
dean-4g             92e8a194-ad60-43c9-a6a1-b861907a00af  wifi      --
elimu-connect       7e379b8f-0caf-4014-a8f9-892134a08e11  wifi      —
```
I modified it to connect to my local SSID and it works, but I don’t have an ElimuConnect here to try. I will try to get another Raspberry Pi tomorrow, but if you can give it a try it would be great 🥺
Thanks!

**Comment by ltoning** [comment]
Hi Alonso,

tested but gives some issues:

TASK [common : Configure WiFi network]
**************************************************************************************************************************************************************************************************************************************************
failed: [elimupi] (item={'name': 'elimu-connect', 'ssid': 'elimu-connect',
'interface': 'wlan0', 'priority': '110', 'psk': '1234567890'}) =>
{"ansible_loop_var": "item", "changed": false, "item": {"interface":
"wlan0", "name": "elimu-connect", "priority": "110", "psk": "1234567890",
"ssid": "elimu-connect"}, "msg": "Error: bad connection type: 'None' not
among [6lowpan, 802-11-olpc-mesh (olpc-mesh), 802-11-wireless (wifi),
802-3-ethernet (ethernet), adsl, bluetooth, bond, bridge, cdma, dummy,
generic, gsm, infiniband, ip-tunnel, loopback, macsec, macvlan, ovs-bridge,
ovs-dpdk, ovs-interface, ovs-patch, ovs-port, pppoe, team, tun, veth, vlan,
vpn, vrf, vxlan, wifi-p2p, wimax, wireguard, wpan, bond-slave,
bridge-slave, team-slave]\n", "name": "elimu-connect", "rc": 2}
failed: [elimupi] (item={'name': 'dean-4g', 'ssid': 'dean-4g', 'interface':
'wlan0', 'priority': '100', 'psk': 'Mobil3internetbyDEAN'}) =>
{"ansible_loop_var": "item", "changed": false, "item": {"interface":
"wlan0", "name": "dean-4g", "priority": "100", "psk":
"Mobil3internetbyDEAN", "ssid": "dean-4g"}, "msg": "Error: bad connection
type: 'None' not among [6lowpan, 802-11-olpc-mesh (olpc-mesh),
802-11-wireless (wifi), 802-3-ethernet (ethernet), adsl, bluetooth, bond,
bridge, cdma, dummy, generic, gsm, infiniband, ip-tunnel, loopback, macsec,
macvlan, ovs-bridge, ovs-dpdk, ovs-interface, ovs-patch, ovs-port, pppoe,
team, tun, veth, vlan, vpn, vrf, vxlan, wifi-p2p, wimax, wireguard, wpan,
bond-slave, bridge-slave, team-slave]\n", "name": "dean-4g", "rc": 2

Op ma 24 jun 2024 om 21:34 schreef Alonso Vidales ***@***.***
>:

> Ahhh, ok, I was not aware of the ElimuConnect and 4G connections, sorry. I
> move them to NetworkManager, now it looks like:
> ```
> ***@***.***:~# nmcli connection show
> NAME UUID TYPE DEVICE
> preconfigured 61f08a09-e3b4-461c-8a32-273b2cd83484 wifi wlan0
> lo 85f3e588-bbf9-465b-9bbb-0816231f9ee0 loopback lo
> Wired connection 1 8ab2b7f0-2545-3145-9394-f2c0edbf7db9 ethernet --
> dean-4g 92e8a194-ad60-43c9-a6a1-b861907a00af wifi --
> elimu-connect 7e379b8f-0caf-4014-a8f9-892134a08e11 wifi —
> ```
> I modified it to connect to my local SSID and it works, but I don’t have
> an ElimuConnect here to try. I will try to get another Raspberry Pi
> tomorrow, but if you can give it a try it would be great 🥺
> Thanks!
>

File: ansible/roles/common/tasks/setup-wifi.yml [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
-    dest: /etc/wpa_supplicant/wpa_supplicant.conf
-    mode: "0400"
+- name: Debug passphrase variables
+  debug:
+    var: wifi_settings.wifi_networks
+
+- name: Configure WiFi network
+  nmcli:
+    conn_name: "{{ item.name }}"
+    state: present
```
Comment (ltoning) [inline ansible/roles/common/tasks/setup-wifi.yml:9]: @alonsovidales  is missing
type: wifi

**Review by ltoning — COMMENTED** [review]
add  type: wifi and then all works fine

File: ansible/roles/common/tasks/setup-wifi.yml [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+    ssid: "{{ item.ssid }}"
+    priority: "{{ item.priority }}"
```
Comment (ltoning) [inline ansible/roles/common/tasks/setup-wifi.yml:11]: priority should be autoconnect-priority

Comment (alonsovidales) [reply ansible/roles/common/tasks/setup-wifi.yml:11]: > priority should be autoconnect-priority

I think that `autoconnect-priority` is not supported by the module: https://docs.ansible.com/ansible/latest/collections/community/general/nmcli_module.html the closest thing is priority but it is a different thing: `This is only used with ‘bridge’ - sets STP priority.`

I'm removing the priority... it should connect to the latest used.

Do we want to keep the `dean-4g` or can we just deprecate it in favour of the connect? 🤔

Comment (ltoning) [reply ansible/roles/common/tasks/setup-wifi.yml]: indeed the module has no support. 
We need it so might something like : 

nmcli connection modify "elimu-connect" connection.autoconnect-priority 110

Comment (alonsovidales) [reply ansible/roles/common/tasks/setup-wifi.yml]: Added as shell 👍

Comment (ltoning) [reply ansible/roles/common/tasks/setup-wifi.yml]: yes this work fine. 
Please finish this with merge.

**Pull request merged**